src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java

```diff
@@ -50,100 +48,69 @@
-        if (region == null) {
-            region = getCurrentRegion(connectionTimeoutSeconds, connectionRetries, readTimeoutSeconds);
-            getLogger().info("Region not set, using the current region: " + region);
-        }
-
-        validateRegion(region);
-
-        final AwsConfig config = AwsConfig.builder().setAccessKey(getOrNull(ACCESS_KEY)).setSecretKey(getOrNull(SECRET_KEY))
-                .setRegion(region)
+    private AwsConfig createAwsConfig() {
+        try {
+            return AwsConfig.builder()
+                .setAccessKey(getOrNull(ACCESS_KEY)).setSecretKey(getOrNull(SECRET_KEY))
+                .setRegion(getOrDefault(REGION.getDefinition(), null))
                 .setIamRole(getOrNull(IAM_ROLE))
                 .setHostHeader(getOrDefault(HOST_HEADER.getDefinition(), DEFAULT_HOST_HEADER))
                 .setSecurityGroupName(getOrNull(SECURITY_GROUP_NAME)).setTagKey(getOrNull(TAG_KEY))
                 .setTagValue(getOrNull(TAG_VALUE))
-                .setConnectionTimeoutSeconds(connectionTimeoutSeconds)
-                .setConnectionRetries(connectionRetries)
-                .setReadTimeoutSeconds(readTimeoutSeconds)
-                .setHzPort(new PortRange(getPortRange())).build();
+                .setConnectionTimeoutSeconds(getOrDefault(CONNECTION_TIMEOUT_SECONDS.getDefinition(),
+                    DEFAULT_CONNECTION_TIMEOUT_SECONDS))
+                .setConnectionRetries(getOrDefault(CONNECTION_RETRIES.getDefinition(), DEFAULT_CONNECTION_RETRIES))
+                .setReadTimeoutSeconds(getOrDefault(READ_TIMEOUT_SECONDS.getDefinition(), DEFAULT_READ_TIMEOUT_SECONDS))
+                .setHzPort(new PortRange(getPortRange()))
+                .build();
 
-        reviewConfiguration(config);
-        return config;
-    }
-
-    String getCurrentRegion(int connectionTimeoutSeconds, int connectionRetries, int readTimeoutSeconds) {
-        String availabilityZone =
-                MetadataUtil.getAvailabilityZone(connectionTimeoutSeconds, connectionRetries, readTimeoutSeconds);
-        return availabilityZone.substring(0, availabilityZone.length() - 1);
-    }
-
-    void validateRegion(String region) {
-        if (!AWS_REGION_PATTERN.matcher(region).matches()) {
-            String message = String.format("The provided region %s is not a valid AWS region.", region);
-            throw new InvalidConfigurationException(message);
+        } catch (IllegalArgumentException e) {
+            throw new InvalidConfigurationException("AWS configuration is not valid", e);
         }
     }
 
     /**
      * Returns port range from properties or default value if the property does not exist.
      * <p>
-     * Note that {@link AbstractDiscoveryStrategy#getOrDefault(PropertyDefinition, Comparable)} cannot be reused, since
+     * Note that {@code AbstractDiscoveryStrategy#getOrDefault(PropertyDefinition, Comparable)} cannot be reused, since
```

Comment:
Why? `@link` makes the reference clickable

Reply:
Right, reverted.